slsim/Deflectors/query_galaxies_as_deflectors.py

```diff
@@ -0,0 +1,408 @@
+#!/usr/bin/env python
+# from math import *
+import numpy as np
+import slsim.srcprop as srcprop
+import slsim.srclensprop as slp
+from astropy.table import Table
+from slsim.Deflectors.light2mass import get_velocity_dispersion
+from uncertainties import unumpy
+from astropy.cosmology import FlatLambdaCDM
+
+##########################################################################################################
+##  This script takes as input a foreground galaxy catalog with RA, DEC, redshift,                      ##
+##  magnitudes, errors, position angle and the ellipticity and uses this information                    ##
+##  to determine which of the galaxies will act as potential lenses, and returns an                     ##
+##  output catalogs of sorted deflectors with the same parameters as in input catalog.                  ##
+##                                                                                                      ##
+##  Note that background source number density is artificially increased to                             ##
+##  increase the chances of lensing due to the respective foreground galaxy.                            ##
+
+##    To use the script in series and parallel:                                                         ##
+##    -- Check the notebook 'example_query_galaxies_as_deflectors.ipynb'                                ##
+##########################################################################################################
+
+
+def find_potential_lenses(
+    foreground_table,
+    background_source,
```

Comment:
As we discussed, we should find potential deflector galaxies (mostly massive elliptical) from the DC2 catalog. Then, you can use these selected deflectors in LensPop.